### PR TITLE
Ticket5 articles by topic

### PR DESCRIPTION
**Pull request by aDiallo21**

### CODE REVIEW

File: src/components/ArticleCard/ArticleCard.jsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
   return (
     <div className="article-card">
       <h3> {article.title}</h3>
-      <p>Topic: {article.topic}</p>
+      <Link className="link" to={`/articles/${article.topic}`}>
+        <p>Topic: {article.topic}</p>
+      </Link>
       <p>Author: {article.author}</p>
       <p> {article.created_at}</p>
       <p> {article.body}</p>
```
Comment (leekli) [inline src/components/ArticleCard/ArticleCard.jsx]: I would probably recommend you don't show the article.body in the all articles section, save this for an individual article page

File: src/components/ArticlesByTopic.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleByTopic } from "../utils/axios";
+import ArticleCard from "./ArticleCard/ArticleCard";
+
+export default function ArticlesByTopic() {
+  const [topicSearch, setTopicSearch] = useState([]);
+  const [loading, isLoading] = useState(true);
```
Comment (leekli) [inline src/components/ArticlesByTopic.jsx]: The second one for useState, by convention should begin with the word 'set'...e.g. setLoading

**Review by leekli — APPROVED** [review]
Thanks Abdoulaye, well done on your ticket.

I am happy to approve this to be merged, I have given 2 bits of feedback though which I would like complete for the next ticket tomorrow, see the comments but the summary is:
* amend the set loading state function name to conventional wording - line 8 in ArticlesByTopic.jsx
* When a user loads the all articles page, or gets all the articles filtered by topic... I would recommend the article.body isn't displayed. You just want show overview info at this point such as: the title, author, date/time.... save the full body text for the individual article page which is the next ticket

Hope that makes sense

**Comment by aDiallo21** [comment]
Thanks for the feedback Lee!

I've removed the body and changed isLoading to setLoading

**Pull request merged**